tests/decodecorpus.c

```diff
@@ -677,11 +695,20 @@ static U32 generateSequences(U32* seed, frame_t* frame, seqStore_t* seqStore,
                     repIndex = MIN(2, offsetCode + 1);
                 }
             }
-        } while (offset > (size_t)((BYTE*)srcPtr - (BYTE*)frame->srcStart) || offset == 0);
+        } while (((!info.useDict) && (offset > (size_t)((BYTE*)srcPtr - (BYTE*)frame->srcStart))) || offset == 0);
 
-        {   size_t j;
+        {
+            size_t j;
+            BYTE* const dictEnd = info.dictContent + info.dictContentSize;
             for (j = 0; j < matchLen; j++) {
-                *srcPtr = *(srcPtr-offset);
+                if ((void*)(srcPtr - offset) < (void*)frame->srcStart) {
```

Comment:
The loop copies a single byte at a time into `srcPtr` and then increments `srcPtr` for a total of `matchLen` times. If a match has overlap with both the dictionary and the current buffer, I believe the `if` condition would be triggered (offset goes past the start of the source), in which case we copy the byte from the dictionary. Once the loop gets to the part of the match that overlaps with the current buffer, the condition is no longer satisfied and it  copies the byte from the source buffer.